src/app/client/src/app/modules/dashboard/components/map/map.component.ts

```diff
@@ -0,0 +1,311 @@
+import { ToasterService } from '@sunbird/shared';
+import { ReportService } from '../../services';
+import { IGeoJSON, ICustomMapObj, Properties, IInputMapData } from '../../interfaces';
+import { Component, OnInit, Input, Output, EventEmitter, ViewChild, AfterViewInit } from '@angular/core';
+import { Map } from 'leaflet';
+import { of, throwError, iif, zip, BehaviorSubject, Subscription } from 'rxjs';
+import { mergeMap, map, retry, catchError, skipWhile, pluck, tap } from 'rxjs/operators';
+import * as mappingConfig from '../../config/nameToCodeMapping.json';
+import { cloneDeep, toLower, find, random, pick } from 'lodash-es';
+
+
+declare var L;
+@Component({
+  selector: 'app-map',
+  templateUrl: './map.component.html',
+  styleUrls: ['./map.component.scss']
+})
+export class MapComponent implements OnInit, AfterViewInit {
+  public mapId;
+  private map: Map;
+  private geoJSONRootLayer;
+  private infoControl;
+  public subscription$;
+  private getGeoJSON = new BehaviorSubject(undefined);
+  @Output() featureClicked = new EventEmitter();
+  private mappingConfig: any;
+  private readonly __defaultConfig = {
+    initialCoordinate: [20, 78],
+    latBounds: [6.4626999, 68.1097],
+    lonBounds: [35.513327, 97.39535869999999],
+    initialZoomLevel: 5,
+    controlTitle: 'India Heat Map',
+    tileLayer: {
+      urlTemplate: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
+      options: {
+        attributions: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+      }
+    },
+    rootStyle: {
```

Comment:
why is the css in the ts file?

Reply:
No, it's not CSS. These are attributions that are required by the leaflet to render its style. By default, I've set open streets but it's configurable and can be changed via the report configuration.
and rest others are default styling configurations passed for the leaflet control box which is also configurable.
If we don't pass the configuration then it'll take these default configurations.